New package: Itera v0.1.0

JuliaRegistrator opened the pull request:
- Registering package: Itera
- Repository: https://github.com/yifen9/Itera.jl
- Created by: @yifen9
- Version: v0.1.0
- Commit: 1354ced6efdc73bed01e92d72e3be90c6d82e443
- Git reference: HEAD

goerz commented:
Thank you for submitting your package! Could you add a little bit of documentation before registering? [At the very least](https://discourse.julialang.org/t/when-does-it-make-sense-to-post-a-new-package-to-the-julia-repository/108701/4), that would be a description of the package's purpose and a small usage example in the README. An important part of packages in General is that any potential user can figure out what the package is about and how to get started with using it. That is really difficult when there is no documentation.

In the longer term, I definitely recommend setting up a [Documenter](https://documenter.juliadocs.org/stable/)-based documentation. Before a `v1.0` release, or for smaller packages that can be effectively described entirely with their `README`, that's not a requirement, though.

GunnarFarneback commented:
[noblock]
Actually there is quite nice documentation in the `docs` directory but it would help discovery if it was linked from the README.

JuliaHubRegistrator committed:
New package: Itera v0.1.0

UUID: e78498a0-16ff-4b72-908b-8a2b3cef9703
Repo: https://github.com/yifen9/Itera.jl.git
Tree: 7c6ea8e838566034e0b2e2b8dd009b8cf1271cdc

Registrator tree SHA: d5716b7a540e5fbc43640c2fff2906fe50e9525a

goerz commented:
I guess those are system prompts for "vibe-coding"? They're pretty good! (And I'm not using "vibe-coding" as a derogatory term here)

So yeah, if you transfer some of the information from those "summary" files in `docs` to the main README (especially the summary and the quick-start snippets), that should be sufficient for registration.

I'd also maybe not keep those system prompt files in `docs` in the long term, since `docs` is usually used for Documenter-based documentation. But whatever works for you. Of course, the existing summary files could serve not just as a starting point for a more complete README, but also for the full documentation, which would then also include API docstrings.

[noblock]

yifen9 commented:
> I guess those are system prompts for "vibe-coding"? They're pretty good! (And I'm not using "vibe-coding" as a derogatory term here)
> 
> So yeah, if you transfer some of the information from those "summary" files in `docs` to the main README (especially the summary and the quick-start snippets), that should be sufficient for registration.
> 
> I'd also maybe not keep those system prompt files in `docs` in the long term, since `docs` is usually used for Documenter-based documentation. But whatever works for you. Of course, the existing summary files could serve not just as a starting point for a more complete README, but also for the full documentation, which would then also include API docstrings.
> 
> [noblock]

The repository is to provide a backend game engine for training AI playing turn-based games (chess, poker, UNO, Pokemon or even Civilization VI etc.).

Thank you for replying! Yes I moved the summary from `docs` to `README.md` but I haven't wrote all the details about the usage of the APIs yet, which I will try to add as soon as possible.

[noblock]
